### PR TITLE
Implement nsmallest and nlargest functions

### LINKED ISSUE #1202: nsmallest function implementation

### LINKED ISSUE #1203: nlargest function implementation

### PR DESCRIPTION
**Pull request by fexolm**
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #1202 Resolves #1203 
- [ ] tests added and passing

### CODE REVIEW

**Comment by fexolm** [comment]
I've benchmarked 2 versions of that function
MapReduceFunction seems 4 times slower then ReduceFunction on every n.

I've checked this on 2xIntel 8276L CPU and 2^23 rows.

Detailed results:
```
%timeit df.nsmallest(5, columns=[0])
MapReduceFunction: 94.2 ms ± 4.51 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
ReduceFunction: 19.7 ms ± 773 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit df.nsmallest(1024, columns=[0])
MapReduceFunction:  91.1 ms ± 1.94 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
ReduceFunction: 19.2 ms ± 651 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit df.nsmallest(2**20, columns=[0])
MapReduceFunction: 91.3 ms ± 4.02 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
ReduceFunction: 20.2 ms ± 423 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -1215,11 +1215,56 @@ def ne(self, other, axis="columns", level=None):
...
+        columns : label or list of labels
+            Column label(s) to order by.
+        keep : {'first', 'last', 'all'}, default 'first'
+            Where there are duplicate values:
+            - `first` : prioritize the first occurrence(s)
+            - `last` : prioritize the last occurrence(s)
+            - ``all`` : do not drop any duplicates, even it means
+                        selecting more than `n` items.
+            .. versionadded:: 0.24.0
         return self._default_to_pandas(pandas.DataFrame.nlargest, n, columns, keep=keep)
```
Comment (ienkovich) [inline modin/pandas/dataframe.py:1239]: Sneaky line

**Comment by gshimansky** [comment]
To run (EXPERIMENTAL) Containerized tests successfully please rebase your PR to current master branch.

File: modin/pandas/dataframe.py (hunk 1 above)
Comment (fexolm) [reply modin/pandas/dataframe.py]: thanks

**Comment by devin-petersohn** [comment]
We can revisit the performance in the future, but `ReduceFunction` is more asynchronous than `MapReduceFunction`, so a true fair comparison would `repr(result)`. For now, we can use this implementation because the performance will be consistent, whereas `MapReduceFunction` will sometimes perform worse.

**Pull request merged**